### DISCUSSION TITLE
"Unable to compare files" result still comes up

### DISCUSSION THREAD

**Discussion by drixoman** [WinMerge/winmerge#636] [General]
Hi, I am coming up with some files being unable to be compared using the latest 2.16.10 x64 version. I tried to google this and it seem to have happened before but was apparently solved and closed. Now this result does come up if the files seems to be corrupted to the point that Windows can't open them (I am making iso from old discs and unfortunately some files seems to have lost their data this way but the file reference are still "there", just that you can't open them with Notepad or Paint and the like.) 

But some of these files reported as "unable to compare" I can open with Notepad (the particular one I was looking at is filled with gibberish/mojibake--seems like some sort of compressed file or perhaps partially corrupted--I can't really tell as it's a program related file with a uncommon file extension, so not a simple text file or image.)

**Comment by sdottaka** [comment]
Are those files in special folders such as Google Drive File Stream folders?

If so, WinMerge currently can't compare files in such special folders, so you might be able to compare them by copying them to another location.

**Comment by chocmake** [comment]
Just wanted to mention that I experienced this error for hundreds of files (out of tens of thousands I was comparing) when the system briefly ran out of memory while WinMerge was comparing directories using binary compare mode on WinMerge 2.16.14.0 (x64).

I was at first puzzled but then remembered the system had ran out of memory for a short period and after re-comparing the directories with sufficient memory the compare was 100% identical unlike the first run.

Thought I'd bring it up since it seems like the error hasn't been reproducible and this might be useful for troubleshooting in the future for someone.

**Comment by eturk1** [comment]
v2.16.30
The failed files are .gdoc, .gsheet, .gdraw

I'm comparing a Google Drive "mirrored" (not streamed) folder with an old copy of the folder look for diffs.

**Reply by sdottaka** [reply]
Unfortunately, files with extensions such as .gdoc, .gsheet, and .gdraw are special files that cannot be opened not only with WinMerge, but also with text and binary editors.

**Comment by boelle** [comment]
i also have this come up.. on mp4 files, but not all of them, some in the same directry are fine... some are not

some that it cant compare compare fine when i right clik and do a compare

**Reply by sdottaka** [reply]
Does "Unable to compare files" still appear when you compare again?

**Comment by boelle** [comment]
yes, but not always

**Reply by sdottaka** [reply]
Please let me know if you know how we can reproduce it.

**Comment by boelle** [comment]
I have no idea

fre. 7. jul. 2023 15.02 skrev Takashi Sawanaka ***@***.***>:

**Comment by ericblade** [comment]
I am seeing this on a fair chunk of files when trying to compare two different WSL images via the \\wsl.localhost\ network mapping.  I can open the files in Windows, but WinMerge just tells me it can't open them.  

Also once it gets within about 5000 files from the end of the 19600 file comparison, it slows down to the point where it's comparison-per-second counter is at 0-0.5, despite it just comparing mostly 100 byte files or less.  I suppose that could be the P9fs throughput tanking, though, but i'm not real sure.

**Reply by busssard** [reply]
same here.

I see it when comparing library files
![grafik](https://github.com/user-attachments/assets/ca27a2c8-b23b-4416-8da6-7b7a07966f3f)

**Reply by sdottaka** [reply]
Are you also having this issue comparing WSL images?
Can you please tell me the full path of the two folders?

**Reply by sdottaka** [reply]
The file or directory that cannot be compared is probably a symbolic link.
This cannot be opened with Notepad etc., so I don't think WinMerge can access it either.

**Reply by busssard** [reply]
@sdottaka yes it makes sense i cannot compare mtab. it is a symlink.
but i would have assumed to be able to compare the libfiles. 

the path is visible.
I created the changes1 folder inside a directory in my home folder, for compiling another embedded system.
Inside there are additions to the output of the build process.

in short the path should not matter, all of these files are in my home directory.

Maybe it would make sense to say why the files cannot be compared?

like an error message:
"Cannot compare symbolic links"
"cannot compare this filetype"
"Unknown error when comparing"

for large images i can at least choose to compare as binary or plain text.
